services/roomService/src/requestHandlers/CoveyTownRequestHandlers.ts

```
@@ -233,3 +261,65 @@ export function townSubscriptionHandler(socket: Socket): void {
     townController.updatePlayerLocation(s.player, movementData);
   });
 }
+
+export async function mazeTimeHandler(): Promise<ResponseEnvelope<MazeCompletionTimeListResponse>> {
+  try {
+    const results = await pool.query(getMazeCompletionTime);
+    return {
+      isOK: true,
+      response: {
+        mazeCompletionTimes: results.rows.map(value => ({
+          playerID: value.player_id,
+          username: value.username,
+          time: value.time,
+        })),
+      },
+    };
+  } catch (error) {
+    return {
+      isOK: false,
```

Might want to return the error message as well